Hummingbird.docc/Articles/ServerProtocol.md

```diff
@@ -0,0 +1,122 @@
+# Server protocol
+
+@Metadata {
+    @PageImage(purpose: icon, source: "logo")
+}
+
+Support for TLS and HTTP2 upgrades
+
+## Overview
+
+By default a Hummingbird application runs with a HTTP/1.1 server. The Hummingbird comes with additional libraries that allow you to change this to use TLS, HTTP2 and WebSockets
+
+### Setting server protocol
+
+When you create your ``Application`` there is a parameter `server` that is used to define the server protocol and its configuration. Below we are creating a server that support HTTP1 with a idle timeout for requests set to one minutes.
+
+```swift
+let app = Application(
+    router: router,
+    server: .http1(idleTimeout: .seconds(60))
+)
+```
+
+## HTTPS/TLS
+
+HTTPS is pretty much a requirement for a server these days. Many people run Nginx in front of their server to implement HTTPS, but it is also possible to setup HTTPS inside your Hummingbird application. 
+
+```swift
+import HummingbirdTLS
+
+let tlsConfiguration = TLSConfiguration.makeServerConfiguration(
+    certificateChain: certificateChain,
+    privateKey: privateKey
+)
+let app = Application(
+    router: router,
+    server: .tls(.http1(), tlsConfiguration: tlsConfiguration)
+)
+```
+
+HTTPS is the HTTP protocol with an added encryption layer of TLS to protect the traffic. The `tls` function applies the encryption layer using the crytographic keys supplied in the `TLSConfiguration`.
+
+## HTTP2
+
+HTTP2 is becoming increasingly common. It allows you to service multiple HTTP requests concurrently over one connection. The HTTP2 protocol does not require you to use TLS but it is in effect only supported over TLS as there aren't any web browsers that support HTTP2 without TLS. Given this the Hummingbird implementation also requires TLS.
```

Comment:
```suggestion
HTTP2 is becoming increasingly common. It allows you to service multiple HTTP requests concurrently over one connection. The HTTP2 protocol is only supported over TLS by both Hummingbird 2 and all web browsers, and as such you don't need to wrap it in TLS yourself.
```

Reply:
I don't want to imply the decision to only support HTTP2 with TLS is a decision made by Hummingbird. It is a decision made by the major browser providers 8 years ago